src/Engine/Trigger/TriggerOnce.cpp

```diff
@@ -0,0 +1,69 @@
+#include "TriggerOnce.hpp"
+
+TriggerOnce::TriggerOnce() {
+
+}
+
+TriggerOnce::~TriggerOnce() {
+
+}
+
+void TriggerOnce::OnEnter() {
+
+}
+
+void TriggerOnce::OnLeave() {
+
+}
+
+void TriggerOnce::OnRemain() {
+
+}
+
+std::string TriggerOnce::GetName() {
+    return name;
+}
+
+void TriggerOnce::SetName(std::string value) {
+    name = value;
+}
+
+std::string TriggerOnce::GetTargetFunction() {
+    return targetFunction;
+}
+
+void TriggerOnce::SetTargetFunction(std::string value) {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Function parameter 'value' should be passed by reference. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://localhost:9000/coding_rules#rule_key=cppcheck%3ApassedByValue)
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') The function 'SetTargetFunction' is never used. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)